src/core/ipns/publisher.js

```diff
@@ -108,13 +108,13 @@ class IpnsPublisher {
       return callback(err)
     }
 
-    // TODO Routing - this should be replaced by a put to the DHT
-    this._repo.datastore.put(key, rec.serialize(), (err, res) => {
+    // Add record to routing (buffer key)
+    this._routing.put(key.toBuffer(), rec.serialize(), (err, res) => {
       if (err) {
         const errMsg = `ipns record for ${key.toString()} could not be stored in the routing`
 
-        log.error(errMsg)
-        return callback(errcode(new Error(errMsg), 'ERR_STORING_IN_DATASTORE'))
+      log.error(errMsg)
+      return callback(errcode(new Error(errMsg), 'ERR_STORING_IN_DATASTORE'))
```

Comment:
Nit: should this error message have a different label? eg ERR_PUTTING_TO_ROUTING?